docs/ActiveLabel.rst

```diff
@@ -0,0 +1,135 @@
+ActiveLabel
+===========
+
+``ActiveNode`` provides an ability to define inheritance of models which also gives subclasess the labels of their parent models.  In Ruby, however, inheritence of classes is not sufficient.  Sometimes is makes more sense to be able to build a module which defines behavior (or "concerns") which could be applied to any model.  This is what ``ActiveLabel`` provides.
+
+``ActiveLabel`` modules can be defined in two ways:
+
+ * Default: Where the module's behavior is always defined on the ``ActiveNode`` model and the model's nodes always have a corresponding label in Neo4j
+ * Optional: Where the module's behavior is defined on the class only when the model's nodes have a corresponding label in Neo4j
+
+.. code-block:: ruby
+
+  class Person
+    include Neo4j::ActiveNode
+
+    property :name, type: String
+
+    label :HasAddress
+    label :Destroyable
+  end
+
+  class Organization
+    include Neo4j::ActiveNode
+
+    property :title, type: String
+
+    label :HasAddress
+    label :Destroyable
+  end
+
+
+.. code-block:: ruby
+
+  class Address
+    property :line1, type: String
+    property :line2, type: String
+    property :country, type: String
+    property :postal_code, type: String
+  end
+
+  module HasAddress
+    include Neo4j::ActiveLabel
+
+    included do
+      has_one :out, :address, type: :HAS_ADDRESS
+    end
+
+    module InstanceMethods
+      def distance_from(has_address_object)
+        address.distance_from(has_address_object.address)
+      end
+    end
+  end
+
+
+.. code-block:: ruby
+
+  module Destroyable
+    include Neo4j::ActiveLabel
+
+    follows_label :Destroyed
+
+    included do
+      property :destroyed_at, type: DateTime
+    end
+
+    module InstanceMethods
+      def destroy
+        destroyed_at = Time.now
+
+        super
+      end
+    end
+
+    module ClassMethods
+      def destroyed_recently
+        all.where("#{identity}.destroyed_at > ?", 1.week.ago)
+      end
+    end
+  end
+
+
+Creating
+--------
+
+If an ``ActiveLabel`` does not declare ``follows_label``, creating a node will attach the corresponding label.  Otherwise you must trigger the attachment of the label:
+
+.. code-block:: ruby
+
+  # Node gets both `Person` and `HasAddress` labels
+  person = Person.create
+
+  # `Destroyed' label is added.  `mark_destroyed` method is automatically defined via `follows_label` definition
+  person.label_as_destroyed
+
+  # `Destroyed' label is removed
+  person.label_as_not_destroyed
```

Comment:
@leehericks Responding to your comment:

I can see what you're saying about the magic prefixes of `label_as_` and `label_as_not_`, though I feel like this is a pretty common Ruby practice.  To me, the act of changing the state of the node to add / remove a label is semantically an action and not setting a property.  Setting a boolean value seems confusing to me because that boolean value isn't actually going anywhere

I definitely see the appeal of the consistent naming (though again, the Ruby convention that I've seen is to generally go with different phrasings of English even if they aren't exactly the same).  Are you saying that we might have `method_missing` magic like `labelled_foo_bar = true` / `label_as_foo_bar` which would add a `FooBar` label even if there wasn't an `ActiveLabel` module?

Reply:
@cheerfulstoic The boolean value needs to go somewhere. Adding and removing labels should fit within the persistence workflow just like properties and be able to check them or enforce some logic in callbacks. If something fails, rolling back the save or update should also make sure the labels are not set. If labels are applied as individual actions, how do you rollback or save errors to the model? This is the same problem I have with associations autosaving right now.